sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala

```diff
@@ -59,9 +59,9 @@ object HiveResult {
         .map(_.mkString("\t"))
   }
 
-  private lazy val zoneId = DateTimeUtils.getZoneId(SQLConf.get.sessionLocalTimeZone)
-  private lazy val dateFormatter = DateFormatter(zoneId)
-  private lazy val timestampFormatter = TimestampFormatter.getFractionFormatter(zoneId)
+  private def zoneId = DateTimeUtils.getZoneId(SQLConf.get.sessionLocalTimeZone)
+  private def dateFormatter = DateFormatter(zoneId)
+  private def timestampFormatter = TimestampFormatter.getFractionFormatter(zoneId)
```

Comment:
@MaxGekk, which codes path access `SQLConf.get` here? Seems like we should clarify in the documentation that we should take other sessions into account since `TimestampFormatter` behaviours can be dependent on SQL configuration when this instance is created..

Reply:
It's SQLConf.get.sessionLocalTimeZone

Reply:
Ah, damn, sure. Yes.